chromadb/auth/__init__.py

```diff
@@ -1,449 +1,176 @@
-        pass
-
-
-# SERVER-SIDE Abstractions
-
-
-class ServerAuthenticationRequest(EnforceOverrides, ABC, Generic[T]):
-    @abstractmethod
-    def get_auth_info(self, auth_info_type: AuthInfoType, auth_info_id: str) -> T:
-        """
-        This method should return the necessary auth info based on the type of
-        authentication (e.g. header, cookie, url) and a given id for the respective
-        auth type (e.g. name of the header, cookie, url param).
-
-        :param auth_info_type: The type of auth info to return
-        :param auth_info_id: The id of the auth info to return
-        :return: The auth info which can be specific to the implementation
-        """
+    def authenticate(self) -> ClientAuthHeaders:
         pass
 
 
-class ServerAuthenticationResponse(EnforceOverrides, ABC):
-    @abstractmethod
-    def success(self) -> bool:
-        ...
-
-    @abstractmethod
-    def get_user_identity(self) -> Optional[UserIdentity]:
-        ...
-
-
-class SimpleServerAuthenticationResponse(ServerAuthenticationResponse):
-    """Simple implementation of ServerAuthenticationResponse"""
-
-    _auth_success: bool
-    _user_identity: Optional[UserIdentity]
-
-    def __init__(
-        self, auth_success: bool, user_identity: Optional[UserIdentity]
-    ) -> None:
-        self._auth_success = auth_success
-        self._user_identity = user_identity
-
-    @override
-    def success(self) -> bool:
-        return self._auth_success
+@dataclass
+class UserIdentity:
+    """
+    UserIdentity represents the identity of a user. In general, not all fields
+    will be populated, and the fields that are populated will depend on the
+    authentication provider.
+    """
+    user_id: str
+    tenant: Optional[str] = None
+    databases: Optional[List[str]] = None
+    # This can be used for any additional auth context which needs to be
+    # propagated from the authentication provider to the authorization
+    # provider.
+    attributes: Optional[Dict[str, Any]] = None
 
-    @override
-    def get_user_identity(self) -> Optional[UserIdentity]:
-        return self._user_identity
 
+class ServerAuthenticationProvider(Component):
+    """
+    ServerAuthenticationProvider is responsible for authenticating requests. If
+    a ServerAuthenticationProvider is configured, it will be called by the
+    server to authenticate requests. If no ServerAuthenticationProvider is
+    configured, all requests will be authenticated.
 
-class ServerAuthProvider(Component):
+    The ServerAuthenticationProvider should return a UserIdentity object if the
+    request is authenticated for use by the ServerAuthorizationProvider.
+    """
     def __init__(self, system: System) -> None:
         super().__init__(system)
+        self._ignore_auth_paths: Dict[
+            str, List[str]
+        ] = system.settings.chroma_server_auth_ignore_paths
+        self.overwrite_singleton_tenant_database_access_from_auth = (
```

Comment:
Never overwrite a given tenant or db -- only None. change client preflight checks to only pass if they're specified, not if default